### PR TITLE
Add default templates option into values.

### PR DESCRIPTION
**Pull request by bianpengyuan**
This is a follow up of #29731. This allows customizing sidecar globally, addresses use cases like https://github.com/istio/istio/issues/29791

### CODE REVIEW

File: operator/cmd/mesh/testdata/manifest-generate/samples/addons/kiali.yaml [hunk 1]
```diff
@@ -0,0 +1,1615 @@
+---
```
Comment (howardjohn) [inline operator/cmd/mesh/testdata/manifest-generate/samples/addons/kiali.yaml:1]: this seems unexpected?

**Review by howardjohn — COMMENTED** [review]
LGTM

File: operator/cmd/mesh/testdata/manifest-generate/samples/addons/kiali.yaml (hunk 1 above)
Comment (bianpengyuan) [reply operator/cmd/mesh/testdata/manifest-generate/samples/addons/kiali.yaml]: Thanks, removed.

**Review by ostromart — APPROVED** [review]

**Pull request merged**